ray-on-gke/tpu/kuberay-tpu-webhook/main.go

```diff
@@ -708,6 +754,59 @@ func init() {
 	klog.InitFlags(nil)
 }
 
+// addPod allows next goroutine to start once the webhook PodInformer cache updates
+func (t *TPUWebhookServer) addPod(obj interface{}) {
```

Comment:
This function has a lot of logic and some delicate conditions to trigger calling wg.Done(), is there a way we can simplify the logic with helper functions? For both readability & risk of breaking the wg calls later

Reply:
Moved all the logic to check if the Pod matches `t.lastAdmitted` to a helper function `isLastAdmittedPod`. I also added a unit test to cover all the cases for addPod. [8f4a349](https://github.com/GoogleCloudPlatform/ai-on-gke/pull/859/commits/8f4a349cc4bc444010085ddcb5bef359c4f154dc)